style/hudl_sql_style_guide.md

```diff
@@ -5,33 +5,101 @@ description: A guide to writing clean, clear, and consistent SQL.
+  - If you do frequent range filtering or equality filtering on one column, specify that column as the sort key. Amazon Redshift can skip reading entire blocks of     data for that column. It can do so because it tracks the minimum and maximum column values stored on each block and can skip blocks that don't apply to the         predicate range.
+  - If you frequently join a table, specify the join column as both the sort key and the distribution key. Doing this enables the query optimizer to choose a sort     merge join instead of a slower hash join. Because the data is already sorted on the join key, the query optimizer can bypass the sort phase of the sort merge       join.
+  
+ * __Dist Key Selection__: When you execute a query, the query optimizer redistributes the rows to the compute nodes as needed to perform any joins and aggregations. The goal in selecting a table distribution style is to minimize the impact of the redistribution step by locating the data where it needs to be before the query is run. Some suggestions for best approach include:
+ 
+    - Distribute the fact table and one dimension table on their common columns. Your fact table can have only one distribution key. Any tables that join on another     key aren't collocated with the fact table. Choose one dimension to   collocate based on how frequently it is joined and the size of the joining rows. Designate     both the dimension table's primary key and the fact table's corresponding foreign key as the DISTKEY.
+    - Choose the largest dimension based on the size of the filtered dataset. Only the rows that are used in the join need to be distributed, so consider the size of     the dataset after filtering, not the size of the table.
+    - Choose a column with high cardinality in the filtered result set. If you distribute a sales table on a date column, for example, you should probably get fairly     even data distribution, unless most of your sales are seasonal. However, if you commonly use a range-restricted predicate to filter for a narrow date period,       most of the filtered rows occur on a limited set of slices and the query workload is skewed.
+    - Change some dimension tables to use ALL distribution. If a dimension table cannot be collocated with the fact table or other important joining tables, you can     improve query performance significantly by distributing the entire table to all of the nodes. Using ALL distribution multiplies storage space requirements and     increases load times and maintenance operations, so you should weigh all factors before choosing ALL distribution.
+
+### Naming Conventions
+
+* Rename and recast fields once. The earlier in the DAG this can be done, the better.
+* Use prefixes in table names (for example, stg_, fct_ and dim_) to indicate which relations should be queried by end users.
```

Comment:
With staging, we started with using suffixes instead of prefixes. Not saying that's right, but there are a number of staging models already using that style.